### ISSUE TITLE
[FEM] scale for result pipelines and filters not updated in view

### ISSUE THREAD

**Issue by donovaly** [FreeCAD/FreeCAD#7230]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?p=610660#p610660

### Version

0.20 (Release)

### Full version info

```shell
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.29177 (Git)
Build type: Release
Branch: releases/FreeCAD-0-20
Hash: 68e337670e227889217652ddac593c93b5e8dc94
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.2
```

### Subproject(s) affected?

FEM

### Issue description

The FEM pipelines suffer a major display issue that its view is not updated:

- assuming you have several pipelines or several filters for a single pipeline
- you select the pipeline or filter in the tree view
result: the corresponding object is shown but the scale displays right besides it still shows the numbers of the former pipeline
This is confusing and also annoying.

Here is an example:
![FreeCAD_nMBoh0LK6l](https://user-images.githubusercontent.com/1828501/180109265-c5b13b34-ee39-40d2-9ca9-28bbf77eca81.gif)

So there are different pipelines for different result eigenmodes. One can see that every one has of course another color distribution but the scale does not change.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by donovaly** [comment]
The main issue is now solved by #7231, but there is still one thing:

![FreeCAD_ZlgBs0FRSd](https://user-images.githubusercontent.com/1828501/180330878-65deec82-1b32-489e-8f7e-3cb519590167.gif)

So when the invisible pipeline is selected in the tree, the color bar is correctly not updated (since we treat the bar as invisible like the object and because we have a visible object and its color bar must fit to it.

But then the invisible pipeline is selected afterwards it is made visible. In case it is made visible, the color bar must be updated to fit to the new shown object.

**Comment by donovaly** [comment]
> The main issue is now solved by #7231, but there is still one thing:

This one was solved by commit 689737b4

**Comment by donovaly** [comment]
There is now still an issue that on hiding an object the color bar stays wrong:

![FreeCAD_0g5xDmEa2j](https://user-images.githubusercontent.com/1828501/180613665-ed6e9261-d110-40b5-935c-83f71e0c06b6.gif)

**Comment by donovaly** [comment]
The remaining part is now fixed too by commit 8e7e05743
